### PR TITLE
[PLUGIN-1577] Changed the logic to use HttpHost constructor for creating AuthScope

### PR DESCRIPTION
**Pull request by MrRahulSharma**
Changed the logic to use HttpHost constructor for creating auth scope in place of its create method.
Cherry-pick : #113
JIRA : https://cdap.atlassian.net/browse/PLUGIN-1577

### CODE REVIEW

**Comment by sumitjnn** [comment]
Please add the Jira id to PR title

File: src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java [hunk 1]
```diff
@@ -107,7 +107,8 @@ private CloseableHttpClient createHttpClient() throws IOException {
     // basic auth
     CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
     if (!Strings.isNullOrEmpty(config.getUsername()) && !Strings.isNullOrEmpty(config.getPassword())) {
-      AuthScope authScope = new AuthScope(HttpHost.create(config.getUrl()));
+      URI uri = URI.create(config.getUrl());
+      AuthScope authScope = new AuthScope(new HttpHost(uri.getHost(), uri.getPort(), uri.getScheme()));
```
Comment (sumitjnn) [inline src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java:111]: Nit: Should we use `HttpHost.create(URI uri)` which seems to be doing the same thing internally?

**Review by sumitjnn — COMMENTED** [review]
Please check if we can add a unit test

**Review by sumitjnn — APPROVED** [review]

File: src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java (hunk 1 above)
Comment (MrRahulSharma) [reply src/main/java/io/cdap/plugin/http/source/common/http/HttpClient.java:111]: I believe there might be a mixup with org.apache.commons.httpclient.HttpHost and org.apache.http.HttpHost(used here) class. org.apache.commons.httpclient.HttpHost have the method with URI as argument but org.apache.http.HttpHost does not.

**Comment by MrRahulSharma** [comment]
> Please check if we can add a unit test

Sure, let me create a follow up item post this change to investigate the addition of UT for this change. Not blocking the cherry-pick on this comment.

**Pull request merged**